Core/Core/Submissions/Submission.swift

```diff
@@ -411,3 +416,11 @@ extension Array where Element == SubmissionType {
 public enum SubmissionWorkflowState: String, Codable {
     case submitted, unsubmitted, graded, pending_review, complete
 }
+
+private func md5(_ string: String) -> String {
+    var digest = [UInt8](repeating: 0, count: Int(CC_MD5_DIGEST_LENGTH))
+    string.data(using: .utf8)?.withUnsafeBytes {
+        _ = CC_MD5($0.baseAddress, UInt32($0.count), &digest)
+    }
+    return digest.map { String(format: "%02x", $0) } .joined()
```

Comment:
Okay alright let's go.

Reply:
iOS 13 has a simpler way to do crypto, but we hadn't switched over yet.

```
string.data(using: .utf8).flatMap { Insecure.MD5.hash(data: $0).map { String(format: "%02x", $0) } .joined() } ?? ""
```